starport/pkg/spn/spn.go

```diff
@@ -302,6 +302,93 @@ type GenesisAccount struct {
 	Coins   types.Coins
 }
 
+// ChainSummary represents the summary of a chain in Genesis module of SPN.
+type ChainSummary struct {
+	ChainID            string
+	Source             string
+	TotalValidators    int
+	ApprovedValidators int
+	TotalProposals     int
+	ApprovedProposals  int
+}
+
+// ChainList lists chain summaries
+func (c *Client) ChainList(ctx context.Context, accountName string) ([]ChainSummary, error) {
+	clientCtx, err := c.buildClientCtx(accountName)
+	if err != nil {
+		return []ChainSummary{}, err
+	}
+
+	// List the chains
+	q := genesistypes.NewQueryClient(clientCtx)
+	req := &genesistypes.QueryListChainsRequest{}
+	chainList, err := q.ListChains(ctx, req)
+	if err != nil {
+		return []ChainSummary{}, err
+	}
+
+	var chainSummaries []ChainSummary
+
+	// Get the summary of each chain
+	for _, chain := range chainList.Chains {
```

Comment:
Can we handle each chain in a different goroutine for faster response times for the CLI? We can use `errgroup` for this.

Reply:
Thanks for the suggestion, the command started to become very slow actually